drivers/dac/ad5754r/ad5754r.c

```diff
@@ -0,0 +1,684 @@
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+/******************************************************************************/
+
+/**
+ * @brief Write device register.
+ * @param dev- The device structure.
+ * @param instr_addr - The instruction address comprising register and
+ *				dac channel address.
+ * @param reg_val - The data to be written.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ad5754r_write(struct ad5754r_dev *dev,
+		  uint8_t instr_addr,
+		  uint16_t reg_val)
+{
+	uint8_t buff[3];
+
+	if (!dev)
+		return -EINVAL;
+
+	/* Instruction frame format - {R/Wb, 0, R2, R1, R0, A2, A1, A0}
+	 * R2, R1, R0 : Register address bits
+	 * D2, D1, D0 : DAC address bits */
+	buff[0] = AD5754R_WRITE | instr_addr;
+	buff[1] = no_os_field_get(AD5754R_BYTE_H, reg_val);
+	buff[2] = no_os_field_get(AD5754R_BYTE_L, reg_val);
+
+	return no_os_spi_write_and_read(dev->spi_desc, buff, sizeof(buff));
+}
+
+/**
+ * @brief Read device register.
+ * @param dev - The device structure.
+ * @param instr_addr - The instruction address comprising register and
+ *				dac channel address.
+ * @param reg_val - The data read from the register.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ad5754r_read(struct ad5754r_dev *dev,
+		 uint8_t instr_addr,
+		 uint16_t *reg_val)
+{
+	int ret;
+	uint8_t buff[3];
+
+	if (!dev || !reg_val)
+		return -EINVAL;
+
+	/* Instruction frame format - {R/Wb, 0, R2, R1, R0, A2, A1, A0}
+	 * R2, R1, R0 : Register address bits
+	 * D2, D1, D0 : DAC address bits */
+	buff[0] = AD5754R_READ | instr_addr;
+
+	ret = no_os_spi_write_and_read(dev->spi_desc, buff, sizeof(buff));
+	if (ret)
+		return ret;
+
+	*reg_val = (buff[1] << 8) | buff[2];
+
+	return ret;
+}
+
+/**
+ * @brief Update specific register bits.
+ * @param dev - The device structure.
+ * @param instr_addr - The instruction address comprising register and
+ *				dac channel address.
+ * @param mask - Specific bits mask.
+ * @param reg_val - The data to be written.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ad5754r_update_bits(struct ad5754r_dev *dev,
+			uint8_t instr_addr,
+			uint16_t mask,
+			uint16_t reg_val)
+{
+	int ret;
+	uint16_t reg_data;
+
+	if (!dev)
+		return -EINVAL;
+
+	ret = ad5754r_read(dev, instr_addr, &reg_data);
+	if (ret)
+		return ret;
+
+	reg_data &= ~mask;
+	reg_data |= reg_val;
+
+	return ad5754r_write(dev, instr_addr, reg_data);
+}
+
+/**
+ * @brief Read specific register bits using supplied mask.
+ * @param dev - The device structure.
+ * @param instr_addr - The instruction address comprising register and
+ *				dac channel address.
+ * @param mask - Specific bits mask.
+ * @param reg_val - The data read from the register.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ad5754r_read_masked(struct ad5754r_dev *dev,
```

Comment:
i'd drop this function since it does not add much value. Just use `no_os_field_get` inline.